### ISSUE TITLE
 Group captures in JSON output

### ISSUE THREAD

**Issue by anka-213** [BrianHicks/tree-grepper#542]
This was [apparently already the case](https://github.com/BrianHicks/tree-grepper/commit/37729b9b9846d6d76c82cb3d4790bb769ef786cc) in a very early version of this program, but was lost when [`src/extractor.rs` was pulled out to its own module](https://github.com/BrianHicks/tree-grepper/commit/fe00e50f246a7902aa6d2d92bb35c6b9edac6e31).

In order to preserve the grouping, we would need to replace this `flat_map` with a `map` and propagate the nested lists forward (similarly to the first commit linked):
https://github.com/BrianHicks/tree-grepper/blob/c61c96770119fa63008a21177f277db6aaa62d2f/src/extractor.rs#L86
We would probably need to take care to not cause a regression leading to showing empty files again, since we might get lists of empty lists.

**Comment by BrianHicks** [comment]
Yes, I think this would be better overall. It's easy to flatten the matches out in some script, but very hard to recover that information.
